### PR TITLE
MVP component documentation for Heading, Form control, Button, and Flash

### PR DESCRIPTION
**Pull request by emilybrick**
Adding a few components for primer.style/design following the [MVP component guidelines format](https://github.com/primer/design/blob/main/content/components/component-documentation-guidelines/component-MVP-documentation-guidelines.md). All new components usage docs are in the main component list on /design (https://github.com/github/primer/issues/1066).

References
- [ADR](https://github.com/github/primer/blob/main/adrs/2022-11-30-documentation-framework.md)
- Part of https://github.com/github/primer/issues/1066

- Component list
  - Heading
  - Form control (MVP)
  - Button 
    - this is a bit more fleshed out than an "MVP" version, although I would like to fully integrate the Descriptive buttons content from /guides/accessibility in the future. Thank you ✨  @langermank for providing the button documentation from Figma 🙌 
  - Flash (MVP)

### CODE REVIEW

**Review by langermank — APPROVED** [review]
So excited to see the button updates come to life! ✨

File: content/components/button.mdx [hunk 1]
```diff
@@ -0,0 +1,342 @@
+---
+title: Button
+componentId: Button
+description: Buttons are used to initiate actions on a page or form.
```
Comment (josepmartins) [inline content/components/button.mdx:4]: nit-pick: If you use the `componentId` the `description` field` isn't necessary

File: content/components/heading.mdx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
+title: Heading
+componentId: Heading
+description: Headings define the hierarchical structure and importance of the content they contain.
```
Comment (josepmartins) [inline content/components/heading.mdx:4]: nit-pick: same here

**Review by josepmartins — APPROVED** [review]
This is great, ship it!!

**Pull request merged**